packages/@react-aria/menu/src/useSubMenuTrigger.ts

```diff
@@ -0,0 +1,244 @@
+        if (!openTimeout.current) {
+          openTimeout.current = setTimeout(() => {
+            onSubmenuOpen();
+          }, 200);
+        }
+      } else if (!isHovered) {
+        cancelOpenTimeout();
+      }
+    }
+  };
+
+  let onBlur = (e) => {
+    // TODO: getting rid of the (!isElementInChildOfActiveScope(e.relatedTarget) part of the check below fixes hovering to open a root unavailable menu item when focus is on a submenu
+    // but breaks the case where the user is hovering over a submenu's submenu trigger child item and then hovers a root menu item (eg hover lvl 1 item 2 -> hover lvl 2 item 3 -> hover lvl 1 item 3)
+    // Ideally we'd be able to track the full menu tree and then check if focus has moved to an element that isn't part of the current submenu tree and close it then
+    if (state.isOpen && parentMenuRef.current.contains(e.relatedTarget)) {
+      onSubMenuClose();
+    }
+  };
+
+  // TODO: Track the external element being interacted with so that we can move focus to it when the subMenu closes. If we don't manually move focus, then focus gets lost to the body
+  // Calling ref.current.focus in useMenuItem's onHoverStart doesn't seem to move focus properly unfortunately
+  let closeTarget = useRef(null);
+  let shouldCloseOnInteractOutside = (target) => {
+    if (target !== ref.current) {
+      closeTarget.current = target;
+      return true;
+    }
+
+    return false;
+  };
+
+  // TODO: this onExit is Spectrum specific, so kinda weird to have in the aria hook. However, perhaps it would be nice if the aria Popover has onExit as well so
+  // users wouldn't need to reimplement this?
+  let onExit = () => {
+    // TODO: A bit awkward, but this first part of the if statement handles moving focus to the item the user hovers, specifically if said item is a menu item
+    // in one of the menu ancestors of the currently open submenu. The ideal logic would be to check if the closeTarget was in any of the menus in the tree and do nothing if so. Technically
+    // the else if part of this handles this already since hovering a submenu item sets it as the focusedKey and then focusing the submenu itself will move focus to the focusedKey via useSelectableCollection,
+    // but it breaks if the user hovers the base menu's first submenu trigger since then the first submenu won't actually close
+    // due to the logic in shouldCloseOnInteractOutside, and thus the first submenu retains focus, overriding the hover focus of the root menu's submenu trigger
+    if (closeTarget.current) {
+      closeTarget.current?.focus();
+      closeTarget.current = null;
+    } else if (!parentMenuRef.current.contains(document.activeElement)) {
+      // need to return focus to the trigger because hitting Esc causes focus to go to the subdialog, which is then unmounted
+      // this leads to blur never being fired nor focus on the body
+      parentMenuRef.current.focus();
+    }
+
+    cancelOpenTimeout();
+  };
```

Comment:
Try to get rid of this onExit by:
1. Add wrapping FocusScope w/ restoreFocus around ContextualHelpTrigger's contents. Might make the `else if` here unneeded
2. Get rid of the FocusScopes around each SubMenu, only have it for root menu. Then have Tab NOT move focus between the ContextualHelpTrigger and the parent trigger. Esc will suffice for now. For submenus, tab no longer needs to move between the two (maybe have it close the menus entirely).

Reply:
Might mean we can get rid of closeTarget ref

Reply:
current attempt at getting rid of onExit is in https://github.com/adobe/react-spectrum/tree/submenu_tab_behavior. Running into a lot of issues with focus behavior getting highjacked by FocusScope or onSelectableCollection tab handling. If we want to have tab move to the preceeding/next focusable element adjacent to the trigger button, ContextualHelpTrigger would need to handle the Tab key for the Dialog as well...